### DISCUSSION TITLE
Safelist pattern for HEX values

### DISCUSSION THREAD

**Discussion by VenoM9078** [tailwindlabs/tailwindcss#15030] [Help]
So there's a scenario where I have a massive array with objects like:

```
{
    direction:
      "bg-[radial-gradient(ellipse_at_left,_var(--tw-gradient-stops))]",
    from: "#43496e",
    via: "#544d7e",
    to: "#65589c",
    gradient: "from-[#43496e] via-[#544d7e] to-[#65589c]",
},
```

And previously, I had this with Tailwind colors i.e

```
{
    direction: "bg-gradient-to-t",
    gradient: "from-yellow-500 via-amber-600 to-orange-700",
},
```

But I solved it with this in my tailwind.config.js:

```
{
      pattern:
        /(from|via|to|bg)-(slate|gray|zinc|neutral|stone|red|orange|amber|yellow|lime|green|emerald|teal|cyan|sky|blue|indigo|violet|purple|fuchsia|pink|rose)-([1-9]00|50|950)|white|black|transparent/,
},
```

And I'm trying this pattern for the arbitrary HEX values too i.e

```
{
      pattern:
        /(from|via|to|bg|text|border)-\[#([0-9a-fA-F]{6}|[0-9a-fA-F]{3})\]/,
},
```
    
But no luck so far - so I wanted to ask - has anyone encountered this issue? I know something like this might not have been possible 2 years ago (as indicated by some other answers on the Internet) but since Tailwind has come so far, is it doable now?

Would love an answer to this, thanks!

**Comment by wongjn** [comment]
Arbitary values don't work in `safelist`. You could consider using CSS variables instead though and set the values via the `style` attribute, like:

```html
<div
  class="… from-[var(--from)] via-[var(--via)] to-[var(--to)]"
  style="--from: {{ data.from }}; --via: {{ data.via }}; --to: {{ data.to }}"
>
```
